src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -79,225 +125,472 @@ const useStyles = makeStyles()(() => ({
+    width: '100%',
+  },
+  formWrapper: {
+    display: 'flex',
+    alignItems: 'center',
+    flexDirection: 'row',
+    flexWrap: 'nowrap',
+    [theme.breakpoints.down('md')]: {
+      flexDirection: 'column',
+      flexWrap: 'wrap',
+    },
+  },
+  formInner: {
+    width: '99%',
+    [theme.breakpoints.up('md')]: {
+      width: '100%',
+      margin: theme.spacing(1),
+    },
+  },
+}));
+
+const ContactAvatar = styled(Avatar)(() => ({
+  width: theme.spacing(4),
+  height: theme.spacing(4),
 }));
 
 interface Props {
   id: string;
   name: string;
+  donations: DonationsType[];
   statusTitle: string;
   statusValue: string;
   amount: number;
   amountCurrency: string;
   frequencyTitle: string;
   frequencyValue: string;
-  hideFunction: (hideId: string) => void;
-  updateFunction: (
-    id: string,
-    change: boolean,
-    status?: string,
-    pledgeCurrency?: string,
-    pledgeAmount?: number,
-    pledgeFrequency?: string,
-  ) => Promise<void>;
+  showModal: (
+    contact: ContactType,
+    message: string,
+    title: string,
+    updateType: UpdateTypeEnum,
+  ) => void;
   statuses: FilterOption[];
   setContactFocus: SetContactFocus;
+  avatar?: string;
 }
 
 const Contact: React.FC<Props> = ({
   id,
   name,
+  donations,
   statusTitle,
   statusValue,
   amount,
   amountCurrency,
   frequencyTitle,
   frequencyValue,
-  hideFunction,
-  updateFunction,
+  showModal,
   statuses,
   setContactFocus,
+  avatar,
 }) => {
-  const [values, setValues] = useState({
-    statusValue: statusValue,
-    amountCurrency: amountCurrency,
-    amount: amount,
-    frequencyValue: frequencyValue,
-  });
+  const { pledgeCurrency: pledgeCurrencies } = useApiConstants() || {};
+  const locale = useLocale();
   const { classes } = useStyles();
   const { t } = useTranslation();
-  const accountListId = useAccountListId();
-  const { push } = useRouter();
-  //TODO: Add button functionality
-  //TODO: Show donation history
 
-  const handleChange = (
-    event:
-      | React.ChangeEvent<HTMLSelectElement>
-      | React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
-    props: string,
-  ): void => {
-    setValues((prevState) => ({ ...prevState, [props]: event.target.value }));
-  };
+  const onSubmit = async ({
+    status,
+    pledgeCurrency,
+    pledgeAmount,
+    pledgeFrequency,
+  }: FormAttributes) => {
+    const modalContact = {
+      id: id,
+      status,
+      name: name,
+      pledgeCurrency,
+      pledgeAmount,
+      pledgeFrequency,
+    };
 
-  const handleContactNameClick = () => {
-    setContactFocus(id);
+    showModal(
+      modalContact,
+      t(`Are you sure you wish to update {{source}} commitment info?`, {
+        source: name,
+      }),
+      t('Update'),
+      UpdateTypeEnum.Change,
+    );
   };
 
+  const appealFormSchema = yup.object({
+    statusValue: yup.string().required('Please select a status'),
+    pledgeCurrency: yup.string(),
+    pledgeAmount: yup.number(),
```

Comment:
I'm not sure what the issue is here. But it wants me to put a number after I select a status. I shouldn't need to put a number since it is not required.
<img width="862" alt="Screenshot 2024-08-09 at 8 23 55 AM" src="https://github.com/user-attachments/assets/711c0e44-13d1-420c-b542-abb70ea44bc0">